### ISSUE TITLE
[Bug] causal_conv1d: wrong gradients when a chunk is shorter than W - 1 tokens

### ISSUE THREAD

**Issue by truong-v** [fla-org/flash-linear-attention#1140]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

The backward routes the final-state gradient `dht` back to the input tokens that produced it. `final_state[d, w]` is input token `t = T - W + w`, so token `t` must receive `dht[d, t + W - T]`. `causal_conv1d_bwd_kernel` derives that slot index from `start_tok = max(0, T - (W - 1))`, and the `max(0, ...)` is the bug: as soon as `T < W - 1` it pins `start_tok` at 0, and every contribution is shifted by `W - 1 - T` slots. For `T >= W - 1` the clamp does nothing and the kernel is correct. fla's own hand-written triton-ascend port of this kernel has the unclamped expression.

A sequence that short also leaves part of `initial_state` still sitting in `final_state` — slot `i_w` survives into `final_state[:, i_w - T]` whenever `i_w >= T` — and `compute_dh0_kernel` drops that term, so `dh0` is wrong under the same condition; the two together are what makes a short chunk's gradients disagree. With the shipped conv width `W = 4` this means chunks of 1 or 2 tokens — single-token decode and short chunked prefill, where the first chunk's `final_state` is handed to the next chunk as its `initial_state`. The forward is exact in these cases, only the gradients are wrong, nothing raises, and the existing backward tests all use `T >= 15`, where the clamp is inert.

https://github.com/fla-org/flash-linear-attention/blob/033b19e81239b13971a410e55dd6c178d430b9d4/fla/modules/conv/triton/kernels.py#L317-L327

The final-state gradient path was added in #568.

### Steps to Reproduce the Bug

fla's own composition contract: running a sequence in two chunks, feeding the first chunk's `final_state` to the second as `initial_state`, must give the same gradients as running the whole sequence at once.

```python
import torch
from fla.modules.conv import causal_conv1d

device, B, T, D, W = 'cuda', 1, 16, 16, 4

def chunked(T1):
    torch.manual_seed(42)
    x = torch.randn(B, T, D, device=device, requires_grad=True)
    weight = torch.randn(D, W, device=device, requires_grad=True)
    bias = torch.randn(D, device=device, requires_grad=True)
    h0 = torch.randn(B, D, W, device=device, requires_grad=True)
    dy = torch.randn(B, T, D, device=device)

    y, _ = causal_conv1d(x, weight, bias, initial_state=h0, activation='silu')
    one_shot = torch.autograd.grad(y, [x, h0], grad_outputs=dy, retain_graph=True)

    y1, state = causal_conv1d(x[:, :T1], weight, bias, initial_state=h0,
                              output_final_state=True, activation='silu')
    y2, _ = causal_conv1d(x[:, T1:], weight, bias, initial_state=state, activation='silu')
    y_chunked = torch.cat([y1, y2], dim=1)
    split = torch.autograd.grad(y_chunked, [x, h0], grad_outputs=dy)

    return ((y_chunked - y).abs().max().item(),
            [(a - b).abs().max().item() for a, b in zip(split, one_shot)])

print(f'{"first chunk":>12} {"forward":>10} {"dx":>10} {"dh0":>10}')
for T1 in [1, 2, 3, 4, 8]:
    fwd, errs = chunked(T1)
    print(f'{T1:>9} tok {fwd:10.2e} {errs[0]:10.4f} {errs[1]:10.4f}')
```

```
 first chunk    forward         dx        dh0
        1 tok   0.00e+00     2.4404     1.3728
        2 tok   0.00e+00     4.4926     1.3728
        3 tok   0.00e+00     0.0000     0.0000
        4 tok   0.00e+00     0.0000     0.0000
        8 tok   0.00e+00     0.0000     0.0000
```

The one-shot call itself matches a plain torch autograd reference to `< 1e-4` at every split, so the disagreement is entirely in the short first chunk's backward, and the break is exactly at `T1 < W - 1 = 3`.

### Expected Behavior

The same gradients whatever the chunk boundaries, as the forward already gives: the slot index taken from the unclamped `T - (W - 1)` (the `(offset >= start_tok) & (offset < T)` mask already keeps the reads in range, and the index stays inside `[1, W)` for every `T >= 1`), and `dh0` also picking up the `dht` slots the chunk was too short to overwrite.

### Environment Information

1. Torch: 2.13.0+cu130
2. Triton: 3.7.1
3. Commit tested: 033b19e8 (current main), Python 3.12, NVIDIA B200, driver CUDA 13.2
